pkg/cli: Add support for uploading debug zip logs to datadog

arjunmahishi opened the pull request:
Currently, the "debug zip upload" is capable of uploading CPU and HEAP
profiles to datadog. This commit extends that by adding support to
upload logs as well.

Datadog doesn't support ingesting historical logs by default. So, we are
working around that by reverse engineering the log rehydration feature.

Datadog has the ability to archive logs after a certain retention period
has passed. These archives live in cloud storage services like S3/GCS
etc. Datadog allows you to "rehydrate" logs from these archives on
demand. This is the [Rehydration feature](https://docs.datadoghq.com/logs/log_configuration/rehydrating/?tab=amazons3).

This commit leverages that ability to ingest historical logs. Instead of
datadog creating the archive, we artificially create an archive with the
logs we want to ingest - using a format that datadog recognises. Then as
mentioned above, we can ingest these logs into datadog by "Rehydrating"
them.

One of the caveats of this approach is, there is no API to trigger
rehydration. It has to be triggered manually on the UI. So, the user
running this automation will see instructions printed on STDERR. This
will contain easy to follow steps to complete the ingestion.

---
Usage

```shell
./cockroach debug zip upload <debug zip dir>  \
  --dd-api-key=<DD API key>) \
  --dd-app-key=<DD APP key> \
  --cluster="<cluster name>" \
  --log-format "crdb-v1" 
``` 

---

Epic: CC-27712
Part of: CC-28517
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/127502)
<!-- Reviewable:end -->

Imran-imtiaz48 reviewed (commented):
The updated code for the debugZipUploadCmd and debugDecodeKeyCmd introduces several enhancements and additional options for better flexibility and specificity in handling debug zip uploads and key decoding. The key changes include the addition of a `dd-site` flag for specifying the Datadog site, new `from` and `to` flags to define a time range for the included artifacts, and a `log-format` flag to specify the format of the input log files. These additions improve the functionality and usability of the commands, allowing for more precise data management and observability. The retained elements ensure backward compatibility and maintain existing functionalities.

Recommendation: Verify the new options in various scenarios to ensure they work as intended and do not introduce unexpected behavior. Additionally, updating documentation to reflect these changes will help users understand and utilize the new capabilities effectively.

dhartunian reviewed (commented):
> Reviewed 1 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi and @arjunmahishi)

___
*[`pkg/cli/debug.go` line 1589 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YtDHWE9jOt-A_cXfG:-O3YtDHX8QfBUGkG3i8O:b-54ue3m) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/debug.go#L1589)):*
> ```Go
> 	f.Var(&debugZipUploadOpts.to, "to", "newest timestamp to include (inclusive)")
> 	f.StringVar(&debugZipUploadOpts.logFormat, "log-format", "",
> 		"log format of the input files")
> ```

don't we have code that infers the format a log file somewhere in `merge-logs`?
___
*[`pkg/cli/zip_upload.go` line 55 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YtJCD1w5_1lwFllgB:-O3YtJCD1w5_1lwFllgC:b-jc6oc) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/zip_upload.go#L55)):*
> ```Go
> 
> 	// the path pattern to search for specific artifacts in the debug
> 	// zip director
> ```

nit: "directory" is what you mean right?
___
*[`pkg/cli/zip_upload.go` line 57 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YtPilEOT4l4EZZ0Oq:-O3YtPilEOT4l4EZZ0Or:bhcwm16) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/zip_upload.go#L57)):*
> ```Go
> 	// zip director
> 	zippedProfilePattern = "nodes/*/*.pprof"
> 	zippedLogsPattern    = "nodes/*/logs/*"
> ```

I don't like hardcoding these path patterns because they prevent someone from importing whatever it is they need. The user should be able to pass a glob or a filename etc. just like they would to any other CLI utility.

Not a blocker.
___
*[`pkg/cli/zip_upload.go` line 135 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YuoGf-1amNJkveezT:-O3YuoGf-1amNJkveezU:b-a5inkj) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/zip_upload.go#L135)):*
> ```Go
> func validateZipUploadReadiness() error {
> 	var (
> 		includeLookup     = map[string]bool{}
> ```

typically sets are implemented using `map[string]struct{}{}` since empty structs take up 0 bytes.
___
*[`pkg/cli/zip_upload.go` line 432 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3Yw2qs4_kTZ_L3t00I:-O3Yw2qs4_kTZ_L3t00J:bzb9mqb) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/zip_upload.go#L432)):*
> ```Go
> 	// wait for all the requests to finish
> 	for i := 0; i < len(reqQueue); i++ {
> 		if err := <-errChan; err != nil {
> ```

I think you might want to use this abstraction here: https://pkg.go.dev/golang.org/x/sync/errgroup


<!-- Sent from Reviewable.io -->

arjunmahishi commented:
*[`pkg/cli/zip_upload.go` line 57 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YtPilEOT4l4EZZ0Oq:-O3ajAh799fSVobgS23S:b-7aezz9) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/zip_upload.go#L57)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

I don't like hardcoding these path patterns because they prevent someone from importing whatever it is they need. The user should be able to pass a glob or a filename etc. just like they would to any other CLI utility.

Not a blocker.
</blockquote></details>

I'd thought about this briefly. I am assuming that the most common use-case will be "here's the debug zip, upload everything we can". For these cases, we have to have hard coded path patterns.

I definitely agree that "importing whatever they need" is useful.

Maybe we can have artifact specific flags like `--logs`, `--profiles` etc along with `--zip`.  If `--zip` is passed, it will use the hard coded paths. if one or more of the specific flags are used, it will user the use provided paths. Thoughts?


<!-- Sent from Reviewable.io -->

arjunmahishi commented:
*[`pkg/cli/zip_upload.go` line 57 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YtPilEOT4l4EZZ0Oq:-O3arOMj2h14cBgE9iqG:bvd8tii) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/zip_upload.go#L57)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

I'd thought about this briefly. I am assuming that the most common use-case will be "here's the debug zip, upload everything we can". For these cases, we have to have hard coded path patterns.

I definitely agree that "importing whatever they need" is useful.

Maybe we can have artifact specific flags like `--logs`, `--profiles` etc along with `--zip`.  If `--zip` is passed, it will use the hard coded paths. if one or more of the specific flags are used, it will user the use provided paths. Thoughts?
</blockquote></details>

This is sounding reasonable to me. Will go ahead and make this change. Let me know you have any other approach in mind. Thanks for pointing this out!


<!-- Sent from Reviewable.io -->

arjunmahishi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi and @dhartunian)

___
*[`pkg/cli/debug.go` line 1589 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YtDHWE9jOt-A_cXfG:-O3aifCfCLrQgKy_Vhls:bbc6ddm) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/debug.go#L1589)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

don't we have code that infers the format a log file somewhere in `merge-logs`?
</blockquote></details>

Nope. I copied this flag from `merge-logs`. 
___
*[`pkg/cli/zip_upload.go` line 55 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YtJCD1w5_1lwFllgB:-O3aiuNr9r5-TImOltmJ:bkhjbmc) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/zip_upload.go#L55)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

nit: "directory" is what you mean right?
</blockquote></details>

Done. Yes, thanks!
___
*[`pkg/cli/zip_upload.go` line 135 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YuoGf-1amNJkveezT:-O3asWOz8jt9MczUG_T4:b-rvd3ad) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/zip_upload.go#L135)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

typically sets are implemented using `map[string]struct{}{}` since empty structs take up 0 bytes.
</blockquote></details>

Done. 
TIL!
___
*[`pkg/cli/zip_upload.go` line 432 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3Yw2qs4_kTZ_L3t00I:-O3aseGKDU6jMIOIejjI:b-v0qj1y) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/zip_upload.go#L432)):*
<details><summary><i>Previously, dhartunian (David Hartunian) wrote…</i></summary><blockquote>

I think you might want to use this abstraction here: https://pkg.go.dev/golang.org/x/sync/errgroup
</blockquote></details>

I was going to use `sync.WaitGroup` but this seems more appropriate. 

This is why I went with the channel approach: 
1. It let's us fail early. With the group approach, we will have to wait until all the batches fail.
2. Since we get a signal on the completion of each batch, we can display a nice progress percentage on STDERR. I think this is pretty useful feedback while waiting for a CLI command finish executing.

Will think more about this. Will try to make the above two points work with `errgroup`.


<!-- Sent from Reviewable.io -->

arjunmahishi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi and @dhartunian)

___
*[`pkg/cli/debug.go` line 1589 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O3YtDHWE9jOt-A_cXfG:-O3pyqi_3XObeTy2nDBn:b3gps75) ([raw file](https://github.com/cockroachdb/cockroach/blob/1dab3d761d783cc26a17f002c045c61a5d000e55/pkg/cli/debug.go#L1589)):*
<details><summary><i>Previously, arjunmahishi (Arjun Mahishi) wrote…</i></summary><blockquote>

Nope. I copied this flag from `merge-logs`.
</blockquote></details>

I was wrong. The format is specified in the header of the log files. I also found a method called `ReadFormatFromLogFile`. Will look into this. 


<!-- Sent from Reviewable.io -->

aa-joshi reviewed (commented):
> Reviewed 2 of 7 files at r1, 6 of 6 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi and @dhartunian)

___
*[`pkg/cli/zip_upload.go` line 165 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qKIMH6GNzDIQnJ9NM:-O5qKIMH6GNzDIQnJ9NN:b-y08iuy) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L165)):*
> ```Go
> 		if _, ok := includeLookup[inc]; ok {
> 			return fmt.Errorf(
> 				"the list of artifacts to upload has duplicates: [%s]", strings.Join(artifactsToUpload, ", "),
> ```

nit: Should we ignore duplicates & continue instead of throwing error?
___
*[`pkg/cli/zip_upload.go` line 250 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qMyuuByDsV86fKPgv:-O5qMyuuByDsV86fKPgw:b-h3w473) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L250)):*
> ```Go
> 
> 			if resp.Body != nil {
> 				defer func() {
> ```

May I know the reason for moving defer function under `if resp.Body != nil ` condition
___
*[`pkg/cli/zip_upload.go` line 267 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qNZz10Kb3FdPHXT5b:-O5qNZz10Kb3FdPHXT5c:b12v100) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L267)):*
> ```Go
> 		}
> 
> 		fmt.Fprintf(os.Stderr, "Uploaded profiles of node %s to datadog (%s)\n", nodeID, strings.Join(paths, ", "))
> ```

Is there any specific reason that we are using `Fprintf` ?
___
*[`pkg/cli/zip_upload.go` line 411 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qSkrxDS0VCJOIKMLW:-O5qSkrxDS0VCJOIKMLX:b-6y3mv) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L411)):*
> ```Go
> 	}
> 
> 	if err := writeLogsToGCS(ctx, chunkMap); err != nil {
> ```

I believe we are holding all logs data in memory here. Should we flush the data to GCS from a given chunk once it cross certain threshold?
Let's set the right expectations with users about data size.
___
*[`pkg/cli/zip_upload.go` line 490 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qTNpb1BxLZDeGNa5o:-O5qTNpb1BxLZDeGNa5p:bsi7kr8) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L490)):*
> ```Go
> 	}
> 
> 	if resp.StatusCode/100 != 2 {
> ```

I see that code is redundant for profile & zip upload. It would be nice if we can omit duplicates.
___
*[`pkg/cli/zip_upload.go` line 524 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qP9gf9C4NGJbMDJpx:-O5qP9gf9C4NGJbMDJpy:bvui0o5) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L524)):*
> ```Go
> // Each path will be uploaded as a separate file. The final file name will be
> // randomly generated just be for uploading.
> var writeLogsToGCS = func(ctx context.Context, chunkMap map[string][][]byte) error {
> ```

Did we explore gcs_storage utility for writing file?
https://github.com/cockroachdb/cockroach/blob/6318706297b37bcec52dd7524661d66b2487a4ab/pkg/cloud/gcp/gcs_storage.go#L1
___
*[`pkg/cli/zip_upload.go` line 543 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qTuU372uRJ3imD9Mc:-O5qTuU372uRJ3imD9Md:b97baec) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L543)):*
> ```Go
> 	}
> 
> 	noOfWorkers := min(system.NumCPU(), len(chunkMap))
> ```

Let's do a benchmarking for optimise gcs upload.


<!-- Sent from Reviewable.io -->

arjunmahishi committed:
pkg/cli: add support for uploading debug zip logs to datadog

Currently, the "debug zip upload" is capable of uploading CPU and HEAP
profiles to datadog. This commit extends that by adding support to
upload logs as well.

Datadog doesn't support ingesting historical logs by default. So, we are
working around that by reverse engineering the log rehydration feature.

Datadog has the ability to archive logs after a certain retention period
has passed. These archives live in cloud storage services like S3/GCS
etc. Datadog allows you to "rehydrate" logs from these archives on
demand. This is the Rehydration feature.

This commit leverages that ability to ingest historical logs. Instead of
datadog creating the archive, we artificially create an archive with the
logs we want to ingest - using a format that datadog recognises. Then as
mentioned above, we can ingest these logs into datadog by "Rehydrating"
them.

One of the caveats of this approach is, there is no API to trigger
rehydration. It has to be triggered manually on the UI. So, the user
running this automation will see instructions printed on STDERR. This
will contain easy to follow steps to complete the ingestion.

Epic: CC-27712
Part of: CC-28517
Release note: None

arjunmahishi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi and @dhartunian)

___
*[`pkg/cli/zip_upload.go` line 165 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qKIMH6GNzDIQnJ9NM:-O5vO8uH6pm9ShU7Gsis:bz4mgmv) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L165)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

nit: Should we ignore duplicates & continue instead of throwing error?
</blockquote></details>

Done.
Makes sense. I don't remember why I decided to throw an error.
___
*[`pkg/cli/zip_upload.go` line 250 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qMyuuByDsV86fKPgv:-O5vOMWmBfT6AfloYNpD:bqwtygg) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L250)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

May I know the reason for moving defer function under `if resp.Body != nil ` condition
</blockquote></details>

This particular request can result is either a `203` response without a body or a `4xx` with a body containing the error message. So, I thought the body will be `nil` in `203` case. 

But I just looked this up. `Body` is never `nil` when `err == nil`. Will fix this.
___
*[`pkg/cli/zip_upload.go` line 267 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qNZz10Kb3FdPHXT5b:-O5vOtO_CjW00MYrCTdo:byioxzd) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L267)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

Is there any specific reason that we are using `Fprintf` ?
</blockquote></details>

To specifically write to `STDERR`. Later when we move this to the pipeline, we might want to use `STDOUT` to print something that will be used as input to the next step in the pipeline. 
___
*[`pkg/cli/zip_upload.go` line 411 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qSkrxDS0VCJOIKMLW:-O5w9Iyt4R5zP560FaLG:b-npn7sp) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L411)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

I believe we are holding all logs data in memory here. Should we flush the data to GCS from a given chunk once it cross certain threshold?
Let's set the right expectations with users about data size.
</blockquote></details>

Yes. Makes sense to change this. I just tried to make the change. I got stuck in resolving a leaky go routine situation. Will create a follow up PR for this. 
___
*[`pkg/cli/zip_upload.go` line 490 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qTNpb1BxLZDeGNa5o:-O5vT4MGDBut3Lya2DOd:bin51k7) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L490)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

I see that code is redundant for profile & zip upload. It would be nice if we can omit duplicates.
</blockquote></details>

Done
Good catch. I've mode the handling of non-2xx into `doUploadReq` itself.
___
*[`pkg/cli/zip_upload.go` line 524 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qP9gf9C4NGJbMDJpx:-O5vWPpbB1a6IBOdOnFY:bpmqiis) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L524)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

Did we explore gcs_storage utility for writing file?
https://github.com/cockroachdb/cockroach/blob/6318706297b37bcec52dd7524661d66b2487a4ab/pkg/cloud/gcp/gcs_storage.go#L1
</blockquote></details>

Yes, I did look into this. The use-case for which it was built seems very different (and it's in the production path). Might have to make significant changes to this code to reuse this. 
___
*[`pkg/cli/zip_upload.go` line 543 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5qTuU372uRJ3imD9Mc:-O5vWkwpD-iiIv5b0v-0:b-2qm5d1) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb8a21f9e20842ef4ef28911b7d64c6b1458496a/pkg/cli/zip_upload.go#L543)):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

Let's do a benchmarking for optimise gcs upload.
</blockquote></details>

Yes. Will do this in the next iteration.


<!-- Sent from Reviewable.io -->

aa-joshi reviewed (changes requested):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arjunmahishi and @dhartunian)

___
*[`-- commits` line 29 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5wM1_s0RiJbDj4PaqO):*
May I know the reason to remove it from commit?


<!-- Sent from Reviewable.io -->

arjunmahishi reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @aa-joshi and @dhartunian)

___
*[`-- commits` line 29 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/127502#-O5wM1_s0RiJbDj4PaqO):*
<details><summary><i>Previously, aa-joshi (Akshay Joshi) wrote…</i></summary><blockquote>

May I know the reason to remove it from commit?
</blockquote></details>

Done
i think I messed this up during --amend


<!-- Sent from Reviewable.io -->

aa-joshi reviewed (approved):
> Reviewed 2 of 3 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)



<!-- Sent from Reviewable.io -->

aa-joshi reviewed (commented):
> Reviewed 1 of 1 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127502)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @dhartunian)



<!-- Sent from Reviewable.io -->

arjunmahishi commented:
bors r+
